app/authoring/integrating-yeoman.md

```diff
@@ -0,0 +1,124 @@
+
+### Find installed generators
+
+But what if you wish to provide access to every Yeoman generators installed on a user machine? Then you need to execute a lookup of the user disk.
+
+```js
+env.lookup(function () {
+  env.run('angular');
+});
+```
+
+`Environment#lookup()` takes a callback that'll be called once Yeoman is done searching for installed generators. Every found generator is going to be registered on the environment.
+
+In case of namespace conflicts, local generators will override global ones.
+
+### Get data about registered generator
+
+Calling `Environment#getGeneratorsMeta()` will return an object describing the meta data the lookup task registered.
+
+Each object keys is a generator namespace, and the value object contains these keys:
+
+- `resolved`: the resolved path to a generator
+- `namespace`: the namespace of the generator
+
+For example:
+
+```json
+{
+  "webapp:app": {
+    "resolved": "/usr/lib/node_modules/generator-webapp/app/index.js",
+    "namespace": "webapp:app"
+  }
+}
+```
+
+Note: Generators registered using `#registerStub()` will have `"unknown"` as `resolved` value.
+
+## Providing a custom User Interface (UI)
+
+Yeoman uses _adapters_ as an abstraction layer to allow IDE, code editor and the like to easily provide user interfaces necessary to run a generator.
+
+An adapter is the object responsible for handling all the interaction with the user. If you want to provide a different interaction model from the classical command line, you have to write your own adapter. Every methods to interact with a user are passing through this adapter (mainly: prompting, logging and diffing).
+
+By default, Yeoman provide a [Terminal Adapter](https://github.com/yeoman/environment/blob/master/lib/adapter.js). And our test helpers provide a [Test Adapter](https://github.com/yeoman/generator/blob/master/lib/test/adapter.js) who's mocking prompts and silencing the output. These are two good examples you can use as reference.
+
+An adapter should provide at least three methods.
+
+### `Adapter#prompt()`
+
+It provides the question-answer functionality (for instance, when you start `yo`, a set of possible actions is prompted to the user). It's signature and behavior follows these of [Inquirer.js](https://github.com/SBoudrias/Inquirer.js). When a generators call `this.prompt`, the call is in the end handled by the adapter.
+
+### `Adapter#diff()`
+Called internally when a conflict is encountered and the user ask for a diff between the old and the new file (both files content is passed as aguments).
+
+### `Adapter#log()`
+It's both a function and an object intended for generic output.
+See [lib/util/log.js](https://github.com/yeoman/environment/blob/master/lib/util/log.js) for the complete list of methods to provide.
+
+## Example implementations
+
+Here's a list of modules/plugins/app using `yeoman-environment`. You can use them as inspiration.
```

Comment:
Yeah it's weird, but we have a new app with a UI maybe coming soon. And maybe some people will come forward with project making use of Yeoman environment.

I'll leave as is for now.